client/jetpack-connect/jetpack-connect-notices.jsx

```diff
@@ -13,6 +13,9 @@ import Notice from 'components/notice';
 
 class JetpackConnectNotices extends Component {
 	static propTypes = {
+		// Supply a function that will be called for flow-ending error cases
+		// instead of showing a notice.
+		onTerminalError: PropTypes.oneOf( [ PropTypes.func, false ] ),
```

Comment:
I think we just need `onTerminalError: PropTypes.func`. It's a nullable function that we want, right?

Reply:
It allows the use of [&&](https://github.com/Automattic/wp-calypso/pull/21568/files#diff-5eb691eb3322d1d81a4e896424f5e7a7R421) but I can change that to a ternary.

Reply:
No problem, I'd missed that bit. `oneOf` is a list of possible values.

I think we need `oneOfType`. If we want to get specific on the `boolean that is false` bit, it would have to be something gnarly and nested like:

```js
PropTypes.oneOfType([ PropTypes.func, PropTypes.oneOf([ false ]) ])
```

But we can probably just

```js
PropTypes.oneOfType([ PropTypes.func, PropTypes.bool ])
```

☝️  untested